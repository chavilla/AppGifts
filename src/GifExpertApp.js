import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import { GridGif } from "./components/GridGif";

const GifExpertApp = () => {
  const [category, setCategory] = useState(["samurai x"]);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategory={setCategory} />
      <hr />

      <ol>
        {category.map((category, i) => (
          <GridGif key={i} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
