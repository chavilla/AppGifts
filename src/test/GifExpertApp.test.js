import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp';


describe('description', () => {

    test('Create the snapshot',()=>{

        const wrapper=shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should get categories', ()=>{
        const categories=['One Punch', 'Chavo'];
        const wrapper=shallow(<GifExpertApp categories={categories}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GridGif').length).toBe(2);

    });

})