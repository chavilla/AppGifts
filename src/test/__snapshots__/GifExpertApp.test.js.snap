// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description Create the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`description should get categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GridGif
      category="One Punch"
      key="0"
    />
    <GridGif
      category="Chavo"
      key="1"
    />
  </ol>
</Fragment>
`;
